client/dashboard/task-list/tasks/payments/payfast.js

```diff
@@ -0,0 +1,177 @@
+      setRequestPending,
+    } = this.props;
+
+    setRequestPending( true );
+
+    await updateOptions( {
+      woocommerce_payfast_settings: {
+        ...this.props.options.woocommerce_payfast_settings,
+        merchant_id: values.merchant_id,
+        merchant_key: values.merchant_key,
+        pass_phrase: values.pass_phrase,
+        enabled: 'yes',
+      }
+    } );
+
+    if ( !isSettingsError ) {
+      recordEvent( 'tasklist_payment_connect_method', {
+        payment_method: 'payfast'
+      } );
+      setRequestPending( false );
+      markConfigured( 'payfast' );
+      createNotice(
+        'success',
+        __( 'PayFast connected successfully', 'woocommerce-admin' )
+      );
+    } else {
+      setRequestPending( false );
+      createNotice(
+        'error',
+        __( 'There was a problem saving your payment setings', 'woocommerce-admin' )
+      );
+    }
+  };
+
+  render() {
+    const helpText = interpolateComponents( {
+      mixedString: __(
+        'Your API details can be obtained from your {{link}}PayFast account{{/link}}',
+        'woocommerce-admin'
+      ),
+      components: {
+        link: <Link 
+          href="https://www.payfast.co.za/"
+          target="_blank"
+          type="external"
+        />,
+      },
+    } );
+
+    return (
+      <Form
+        initialValues={ this.getInitialConfigValues() }
+        onSubmitCallback={ this.updateSettings }
+        validate={ this.validate }
+      >
+        { ( { getInputProps, handleSubmit } ) => {
+          return (
+            <Fragment>
+              <TextControl 
+                label={ __(
+                  'Merchant ID',
+                  'woocommerce-admin'
+                ) }
+                required
+                { ...getInputProps( 'merchant_id' ) }
+              />
+              <TextControl 
+                label={ __(
+                  'Merchant Key',
+                  'woocommerce-admin'
+                ) }
+                required
+                { ...getInputProps( 'merchant_key' ) }
+              />
+              <TextControl 
+                label={ __(
+                  'Passphrase',
+                  'woocommerce-admin'
+                ) }
+                required
+                { ...getInputProps( 'pass_phrase' ) }
+              />
+              <Button onClick={ handleSubmit } isPrimary>
+                { __( 'Proceed', 'woocommerce-admin' ) }
+              </Button>
+              <Button onClick={ () => {
+                this.props.markConfigured( 'payfast' )
+              } }>
+                { __( 'Skip', 'woocommerce-admin' ) }
+              </Button>
+              <p>{ helpText }</p>
+            </Fragment>
+          );
+        } }
+      </Form>
+    );
+  }
+}
+
+export default compose(
+  withSelect( ( select ) => {
+    const { getOptions } = select( 'wc-api' );
+    const options = getOptions( [ 'woocommerce_payfast_settings' ] );
+
+    return {
+      options
```

Comment:
Are we using this prop anywhere?

Reply:
No, I'm still getting comfortable with the pattern and copied this from another payment method.